include/flexflow/simulator.h

```diff
@@ -50,11 +50,22 @@ class FFModel;
 
 struct CostMetrics {
   CostMetrics()
-      : forward_time(0.0f), backward_time(0.0f), sync_time(0.0f),
-        memory_requirement(0) {}
+      : forward_time{}, backward_time{}, sync_time{}, inputs_memory{},
+        outputs_memory{}, weights_memory{} {}
+
+  inline size_t total_memory() const {
+    return inputs_memory + outputs_memory + weights_memory;
+  }
+
+  // Data Members
```

Comment:
```suggestion
public:
```

Just to closer match the rest of the codebase